src/components/util/analytics.js

```diff
@@ -367,61 +354,56 @@ export function eventObjectFromCtx(context){
- * @param {string} [fields.name] - Name of Item we're on, if any.
- * @param {string} [fields.field] - Name of field being acted on, if any.
- * @param {string} [fields.term] - Name of term being acted on or changed, if any.
+ * @param {string} name - event name
+ * @param {string} source - component/page where the event is triggerred
+ * @param {string} action - action that triggers the event
+ * @param {callback} callback - callback function
+ * @param {string} [parameters.filters] -
+ * @param {number} [parameters.experiment_type] -
+ * @param {Object} [parameters.user_groups] -
+ * @param {string} [parameters.term_key] -
+ * @param {string} [parameters.field_key] -.
+ * @param {string} [parameters.value] -
+ * @param {string} [parameters.name] -
+ * @param {string} [parameters.list_name] -
+ * @param {string} [parameters.user_uuid] -
+ * @param {string} [parameters.lab] -
+ * @param {string} [parameters.file_type] -
+ * @param {string} [parameters.file_classification] -
  */
-export function event(category, action, fields = {}, useTimeout = true){
+export function event(name, source, action, callback, parameters = {}, useTimeout = true){
     if (!shouldTrack()) return false;
 
-    const eventObj = _.extend({}, fields, {
-        'hitType'       : 'event',
-        'eventCategory' : category,
-        'eventAction'   : action
-    });
-
-    // Convert internal dimension names to Google Analytics ones.
-    _.pairs(eventObj).forEach(function([key, value]){
-        if (typeof state.dimensionNameMap[key] !== 'undefined'){
-            eventObj["dimension" + state.dimensionNameMap[key]] = value;
-            delete eventObj[key];
-        } else if (typeof state.metricNameMap[key] !== 'undefined'){
-            eventObj["metric" + state.metricNameMap[key]] = value;
-            delete eventObj[key];
-        }
-    });
-
-    eventObj.hitCallback = function(){
-        console.info('Successfuly sent UI event.', eventObj);
+    const defaultCallback = function () {
+        console.info('Successfuly sent UI event. ', arguments);
     };
+    const passParameters = _.pick(
+        _.extend({ source, action, event_callback: callback || defaultCallback }, parameters),
+        function (value) {
+            return typeof value !== 'undefined' && value !== null;
+        });
 
     if (useTimeout){
         setTimeout(function(){
-            ga2('send', eventObj);
+            ga4('event', name, passParameters);
         }, 0);
     } else {
-        ga2('send', eventObj);
+        ga4('event', name, passParameters);
     }
 }
 
 export function setUserID(userUUID){
-    if (!shouldTrack()) {
-        return false;
-    }
-    ga2('set', 'userId', userUUID);
-    console.info("Set analytics user id to", userUUID);
-    return true;
+    // if (!shouldTrack()) {
+    //     return false;
+    // }
+    // ga4('set', 'userId', userUUID);
+    // console.info("Set analytics user id to", userUUID);
+    // return true;
```

Comment:
@Bianca-Morris  Cannot decide whether to send user info since Google warns about GDPR-like restrictions. So, we leave it there as commented out until it gets clear.